File: pac4j-async-core/src/test/java/org/pac4j/async/core/TestProfile.java
package org.pac4j.async.core;

import org.pac4j.core.profile.CommonProfile;

/**
 * Trivial profile for some very simple testing of core infrastructure
 */
public class TestProfile extends CommonProfile {

    public static TestProfile from(final TestCredentials credentials) {
        return new TestProfile(credentials.getName());
    }

    private TestProfile(String name) {
        this.setId(name);
    }

    @Override
    public boolean equals(Object that) {
        if (!(that instanceof TestProfile)) return false;
        final TestProfile other = (TestProfile) that;
        return this.getId() == other.getId();
    }
}

File: vertx-pac4j-async-demo/src/test/kotlin/org/pac4j/async/vertx/TestProfile.kt
package org.pac4j.async.vertx

import org.pac4j.core.profile.CommonProfile

/**
 *
 */
class TestProfile(name: String): CommonProfile() {

    companion object {
        fun from(credentials: TestCredentials): TestProfile {
            return TestProfile(credentials.name)
        }
    }

    init {
        setId(name)
    }

    override fun equals(that: Any?): Boolean {
        if (that !is TestProfile) return false
        val other = that as TestProfile?
        return this.id === other!!.id
    }

}